backend/src/main/scala/bloop/Compiler.scala

```diff
@@ -934,6 +941,8 @@ object Compiler {
                   syntax.replace(readOnlyClassesDirPath, clientClassesDirPath)
                 )
                 if (rebasedFile.exists) {
+                  // In practice this deletes previous successful compilation artifacts
+                  // (like '.tasty' and ".class"), which we do not need in clientDir
```

Comment:
Why wouldn't we need class files in client directory? That would be where metals reads them from no?

Reply:
Yes, but after a best effort compilation with compilation errors (producing only betasty), metals would only need to read betasty there - other files (coming from readOnly) would be outdated at that point and would contain outdated symbols. This is the part of the code I was looking for after the last review, after incorrectly assuming clientDir participated in incremental compilation, so I chose it fit to add this comment here (since I was confused where that was happening)